easybuild/easyconfigs/p/Python/Python-3.6.3-icc_segfault_workaround.patch

```diff
@@ -0,0 +1,60 @@
+This patch was provided by Andrew Edmondson in an email on the EasyBuild mailing list February 28, 2018.
```

Comment:
@schiotz Please include https://lists.ugent.be/wws/arc/easybuild/2018-02/msg00043.html in the comment to point to the actual message.

What about the comment that this requires `-O0`? Maybe @edmondac can provide some more info on this?

Reply:
I mentioned the patch upstream, see https://bugs.python.org/issue33174, maybe they can provide some feedback on this.

Reply:
My email said: 

> I've just experienced this problem too. It seems to be a compiler bug with recent icc (it's still there in 2018.1.163). Python 3.6.x now includes sha3 which used to be a pypi module - and there's one tiny bit of code in there that creates a huge line in C using a macro. Icc segfaults on that line, presumably because it can't fit it in a buffer somewhere.
> 
> Anyway, I've made a patch (attached) which I'll be contributing back along with a Python-3.6.3-iomkl-2018a.eb in the next couple of weeks. If anyone fancies sanity checking the patch then I'd be very grateful! It does make some assumptions about which version of the macro to use (which it just unpacks explicitly into several lines) so someone might have a better idea. Even then, it still needs '-O0' to build it...

The -O0 was just found experimentally.

Reply:
There's some feedback on your patch already, see https://bugs.python.org/issue33174#msg319943 .

In my experience the `-O0` isn't needed at all when the patch is included, at least not with `intel/2018.02` (but then again, I wasn't seeing this problem in the first place, for some reason unclear to me...).

cc @vanzod (who has also run into this)

Reply:
See https://software.intel.com/en-us/forums/intel-c-compiler/topic/780574

Reply:
@boegel The patch itself enables -O0 for the relevant file, so nothing extra needs to be done.

Reply:
I am now testing if -O0 is necessary for me.  Due to a PhD defence, it is possible that I will not get time to reply back until tomorrow.

Reply:
@edmondac Unfortunately, the link you provides is to a closed site.  Even registering does not give access to it :-(   But I assume it means that Intel is working on it.

Reply:
@boegel It *is* necessary to use -O0 to compile the relevant file.  I am uploading the new version with more comments in the patch file.